### ISSUE TITLE
Deleting last path component on Linux can result in URL with empty path

### ISSUE THREAD

**Issue by ahoppen** [swiftlang/swift-foundation#980]
```swift
var url = URL(fileURLWithPath: "/whatever/test.swift")
print("'\(url.path)'") // prints '/whatever/test.swift'
url.deleteLastPathComponent()
print("'\(url.path)'") // prints '/whatever'
url.deleteLastPathComponent()
print("'\(url.path)'") // prints ''

url = URL(fileURLWithPath: "/test.swift")
print("'\(url.path)'") // prints '/test.swift'
url.deleteLastPathComponent()
print("'\(url.path)'") // prints '/'
```

Where it prints `''` at the end of the first section, I think it should print `/`.

This is in a Swift 6.0 Linux docker image.

**Comment by jrflat** [comment]
This `String.deletingLastPathComponent()` bug was fixed in `main` by #897. The implementation was updated in `main` (still fixes the bug) in #927. Neither of those were cherry-picked to `release/6.0`, though.

@parkera Can we cherry-pick #927 (String+Path parsing performance improvements) to `release/6.0`, or should we just take the minimal fix from #897?

**Comment by parkera** [comment]
I think I'd rather take the full patch than attempt to cut it down.
